src/test/java/io/github/jhipster/online/service/JdlServiceTest.java

```diff
@@ -0,0 +1,195 @@
+class JdlServiceTest {
+
+    @Mock
+    private LogsService logsService;
+
+    @Mock
+    private GitService gitService;
+
+    @Mock
+    private JHipsterService jHipsterService;
+
+    @Mock
+    private GithubService githubService;
+
+    @Mock
+    private GitlabService gitlabService;
+
+    @Mock
+    private JdlRepository jdlRepository;
+
+    @Autowired
+    private ApplicationProperties applicationProperties;
+
+    private JdlService jdlService;
+
+    @BeforeEach
+    void shouldConstructJHipsterService() {
+        jdlService =
+            new JdlService(logsService, gitService, jHipsterService, githubService, gitlabService, jdlRepository, applicationProperties);
+    }
+
+    @Test
+    void applyJdlWithException() {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+
+        jdlService.applyJdl(user, "jhipster", "online", jdlMetadata, applyJdlId, GitProvider.GITHUB);
+
+        verify(logsService).addLog(applyJdlId, "Error during generation: Error creating file jhipster-jdl.jh, could not write the file");
+        verify(logsService).addLog(applyJdlId, "Generation failed");
+    }
+
+    @Test
+    void applyGithubJdl() throws GitAPIException, IOException {
+        final User user = new User();
+        final String applyJdlId = "apply-id";
+        final JdlMetadata jdlMetadata = new JdlMetadata();
+        jdlMetadata.setId("1");
+        Jdl jdl = mock(Jdl.class);
+        jdlMetadata.setName("jdl-metadata-name");
+        given(jdlRepository.findOneByJdlMetadataId(jdlMetadata.getId())).willReturn(Optional.of(jdl));
+        given(githubService.getHost()).willReturn(applicationProperties.getGithub().getHost());
+
+        jdlService.applyJdl(user, "jhipster", "online", jdlMetadata, applyJdlId, GitProvider.GITHUB);
+
+        verifyApplyJdl(applyJdlId, jdlMetadata);
+        verify(logsService).addLog(applyJdlId, "Cloning GitHub repository `jhipster/online`");
+        verify(gitService).cloneRepository(eq(user), any(File.class), eq("jhipster"), eq("online"), eq(GitProvider.GITHUB));
+        verify(logsService).addLog(applyJdlId, "Pushing the application to the GitHub remote repository");
+        verify(gitService).push(nullable(Git.class), any(File.class), eq(user), eq("jhipster"), eq("online"), eq(GitProvider.GITHUB));
+        verify(logsService).addLog(applyJdlId, "Creating Pull Request");
+        verify(githubService)
+            .createPullRequest(
+                user,
+                "jhipster",
+                "online",
+                "Add entities using the JDL model `jdl-metadata-name`",
+                "jhipster-entities-apply-id",
+                "Entities generated by JHipster using the model at https://start.jhipster.tech/jdl-studio/#!/view/1"
+            );
+        verify(logsService).addLog(applyJdlId, "Pull Request created at https://github.test/jhipster/online/pull/0");
```

Comment:
Personally I'm not in favor of changing this as you can now query the string (using find in an IDEA) and will have one single reference from src/test/resources application.yml. If we change the assertion to the value from the already wired properties, the test would still succeed if you don't overwrite the file, as it would than check the default value in the source code itself. @SudharakaP  your thoughts?

Reply:
But isn't the whole point of the test is to verify that the log `Pull Request created at <hostname>/jhipster/online/pull/0` is added? So I suppose the host name part we don't care which is why I thought of making it configurable. Whether it comes from the properties file or if it's not overridden there if it comes from the default value in source code doesn't matter since what we are checking for is the remainder of the string right? Am I missing something? 🤔

Reply:
I can follow that thought process as well, I'll update it